app/controllers/monthlyreturns/SubmittedNoReceiptController.scala

```diff
@@ -0,0 +1,110 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.monthlyreturns
+
+import controllers.actions.*
+import models.EmployerReference
+import pages.monthlyreturns.{CisIdPage, ConfirmEmailAddressPage, ContractorNamePage, DateConfirmNilPaymentsPage}
+import play.api.Logging
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.MonthlyReturnService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+import views.html.monthlyreturns.SubmittedNoReceiptView
+import java.time.format.DateTimeFormatter
+import java.time.{Clock, ZoneId, ZonedDateTime}
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SubmittedNoReceiptController @Inject()(
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  requireCisId: CisIdRequiredAction,
+  val controllerComponents: MessagesControllerComponents,
+  view: SubmittedNoReceiptView,
+  clock: Clock,
+  monthlyReturnService: MonthlyReturnService
+) (implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with Logging {
+
+
+  private def formatEmployerRef(er: EmployerReference): String =
+    s"${er.taxOfficeNumber}/${er.taxOfficeReference}"
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData andThen requireCisId).async {
+    implicit request =>
+      implicit val hc: HeaderCarrier =
+        HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+
+      val cisId = request.userAnswers.get(CisIdPage).getOrElse {
+        logger.error("[SubmissionSuccess] cisId missing from userAnswers")
+        throw new IllegalStateException("cisId missing from userAnswers")
+      }
+
+      val contractorName = request.userAnswers.get(ContractorNamePage).getOrElse {
+        logger.error("[SubmissionSuccess] contractorName missing from userAnswers")
+        throw new IllegalStateException("contractorName missing from userAnswers")
+      }
+
+      val employerRef: String =
+        request.employerReference
+          .map(formatEmployerRef)
+          .getOrElse {
+            val msg = s"SubmissionSuccess: employerReference missing for userId=${request.userId}"
+            logger.error(msg)
+            throw new IllegalStateException(msg)
```

Comment:
Where will this go?
Maybe we can redirect to system error controller here instead.